os-server/src/main/java/app/openschool/usermanagement/UserService.java

```diff
@@ -0,0 +1,17 @@
+package app.openschool.usermanagement;
+
+import app.openschool.usermanagement.api.dto.UserRegistrationDto;
+import app.openschool.usermanagement.api.dto.UserRegistrationHttpResponse;
+import app.openschool.usermanagement.entities.RoleEntity;
+import app.openschool.usermanagement.entities.UserEntity;
+import org.springframework.http.ResponseEntity;
+
+/** Useful Javadoc. */
+public interface UserService {
+
+  ResponseEntity<UserRegistrationHttpResponse> register(UserRegistrationDto userDto);
```

Comment:
User Service shouldn't know anything about the higher-level abstractions, please do not use ResponseEntity here.